ext/socket/raddrinfo.c

```diff
@@ -369,14 +369,14 @@ rb_getaddrinfo_a(const char *node, const char *service,
 	arg.timeout = timeout;
 
 	ret = (int)(VALUE)rb_thread_call_without_gvl(nogvl_gai_suspend, &arg, RUBY_UBF_IO, 0);
-
-	if (ret) {
-	    /* on Ubuntu 18.04 (or other systems), gai_suspend(3) returns EAI_SYSTEM/ENOENT on timeout */
-	    if (ret == EAI_SYSTEM && errno == ENOENT) {
-		return EAI_AGAIN;
-	    } else {
-		return ret;
-	    }
+	if (ret && ret != EAI_ALLDONE) {
+		/* EAI_ALLDONE indicates that the request already completed and gai_suspend was redundant */
+		/* on Ubuntu 18.04 (or other systems), gai_suspend(3) returns EAI_SYSTEM/ENOENT on timeout */
+		if (ret == EAI_SYSTEM && errno == ENOENT) {
```

Comment:
I have feeling, indentation among the file is 4 spaces, and here I see 1 tab instad.